rfcs/0000-unified-configuration.ftd

```diff
@@ -0,0 +1,188 @@
+-- import: fastn.com/rfcs/lib
+-- import: fastn-community.github.io/bling/note
+
+-- lib.rfc: RFC: Unified Configuration
+id: unified-config
+status: proposal
+
+TODO: Write a brief summary of the RFC here.
+
+
+-- lib.motivation:
+
+;; TODO: add links to the following paragraph
+Currently, `fastn` supports configuration through environment variables, and in
+some way, through the `FASTN.ftd` file. Environment variables are good to
+change the behaviour of the cli or to pass secret information but it's not
+ideal for defining complex configuration for a framework like `fastn` that
+provides various configurable features like processors, oauth authentication
+and, apps.
+
+;; TODO: I can do better than this
+This RFC proposes a way of dealing with configuration that is centralised and
+leveraged the existing ability of the `fastn` cli to read environment
+variables.
+
+-- end: lib.motivation
+
+
+-- lib.detailed-design:
+
+-- ds.h2: `fastn/config` - A virtual module to assist configuration definition.
```

Comment:
Why are you calling it `fastn/config`, it is `<package-name>/-/config`.

Reply:
Reading more it seems you did not mean that. You have not made a case for why `fastn/config` should exist, that deviates from our `ftd.*` builtin. All config related data types should be under the `fastn.` namespace, fastn only exists for configuration.

Reply:
After thinking about it a bit more, I think `<package-name>/-/config` makes more sense as it's supposed to contain user defined configuration (variables in `FASTN.ftd`).

I'll change the proposal to reflect this and will also mention that the types (eg, `db-conn`) are under `fastn.*` namespace.